module/modmanager/manager_test.go

```diff
@@ -1859,3 +1859,120 @@ func TestGetAutomaticPort(t *testing.T) {
 		test.That(t, lis.Close(), test.ShouldBeNil)
 	}
 }
+
+func TestFailedModuleTracking(t *testing.T) {
+	os.Setenv("VIAM_TCP_SOCKETS", "yes")
+	t.Cleanup(func() { os.Unsetenv("VIAM_TCP_SOCKETS") })
+
+	ctx := context.Background()
+	logger := logging.NewTestLogger(t)
+
+	parentAddr := setupSocketWithRobot(t)
+	opts := modmanageroptions.Options{UntrustedEnv: false}
+	mgr := setupModManager(t, ctx, parentAddr, logger, opts)
+
+	// Add a working module
+	workingMod := []config.Module{
+		{
+			Name:    "working-module",
+			ExePath: rtestutils.BuildTempModule(t, "module/testmodule"),
+			Type:    config.ModuleTypeLocal,
+		},
+	}
+	err := mgr.Add(ctx, workingMod...)
+	test.That(t, err, test.ShouldBeNil)
+
+	// Add two failing modules (invalid exe paths)
+	failingMod1 := []config.Module{
+		{
+			Name:    "failing-module-1",
+			ExePath: "/nonexistent/path/to/module1",
+			Type:    config.ModuleTypeLocal,
+		},
+	}
+	err = mgr.Add(ctx, failingMod1...)
+	test.That(t, err, test.ShouldNotBeNil)
+
+	failingMod2 := []config.Module{
+		{
+			Name:    "failing-module-2",
+			ExePath: "/nonexistent/path/to/module2",
+			Type:    config.ModuleTypeLocal,
+		},
+	}
+	err = mgr.Add(ctx, failingMod2...)
+	test.That(t, err, test.ShouldNotBeNil)
+
+	t.Run("FailedModules returns failing modules properly", func(t *testing.T) {
+		failedModules := mgr.FailedModules()
+		test.That(t, failedModules, test.ShouldContain, "failing-module-1")
```

Comment:
use test.ShouldResemble instead when we know what the failedModule list should look like??